Description: Use renamed factory library from Debian singular package.
 See [1].  We replace the previous configure checks with pkg-config.
 .
 [1] http://anonscm.debian.org/cgit/debian-science/packages/singular.git/tree/
     debian/patches/debianization-rename-libraries.patch
Author: Doug Torrance <dtorrance@piedmont.edu>
Forwarded: https://github.com/Macaulay2/M2/pull/337
Last-Update: 2015-09-01

--- a/M2/configure.ac
+++ b/M2/configure.ac
@@ -1267,45 +1267,23 @@
 
 BUILD_gftables=yes		dnl it seems that factory now includes gftables, and installs it
 
-if test "$DEBUG" = yes
-then SINGULARLIBS="-lfactory  "
-else SINGULARLIBS="-lfactory  "
+SINGULARLIBS="-lfactory  "
+if test $BUILD_factory = no
+then
+   PKG_CHECK_MODULES(FACTORY, factory >= 4.0.0, , BUILD_factory=maybe)
+   if test $BUILD_factory = maybe
+   then
+	BUILD_factory=no
+	PKG_CHECK_MODULES(FACTORY, singular-factory >= 4.0.0, , BUILD_factory=yes)
+   fi
+fi
+if test $BUILD_factory = no
+then
+     LIBS="$FACTORY_LIBS $LIBS"
+     CPPFLAGS="$FACTORY_CFLAGS $CPPFLAGS"
+else
+     BUILTLIBS="$SINGULARLIBS $BUILTLIBS"
 fi
-while test $BUILD_factory = no 
-do AC_LANG(C++)
-   AC_CHECK_HEADER(factory/factory.h,,BUILD_factory=yes; break)
-   SAVELIBS=$LIBS
-   LIBS="$SINGULARLIBS $LIBS"
-   AC_MSG_CHECKING([whether factory library is installed ($SINGULARLIBS)])
-   dnl if -lntl is needed, it's been added to $LIBS above
-   AC_LINK_IFELSE([AC_LANG_SOURCE([[ #undef NOSTREAMIO
-	              #include <factory/factory.h>
-		      int main () { return 0 ; } ]])],
-       [AC_MSG_RESULT(yes)],
-       [AC_MSG_RESULT([no, will build factory]) ; BUILD_factory=yes  LIBS=$SAVELIBS ; break] )
-   AC_MSG_CHECKING([whether up-to-date factory libraries are installed])
-   AC_RUN_IFELSE([AC_LANG_SOURCE([[
-	   #include <stdio.h>
-	   #include <string.h>
-           #undef NOSTREAMIO
-	   #undef PACKAGE_VERSION /* so we can get PACKAGE_VERSION from factoryconf.h */
-	   #include <factory/factory.h>
-	   int main () {
-		FILE *msg = fdopen(]]AS_MESSAGE_FD[[,"w");
-		static int factoryV[3], fV[3] = {4,0,0}, fOK;
-		sscanf(PACKAGE_VERSION,"%d.%d.%d", &factoryV[0], &factoryV[1], &factoryV[2]);
-		fOK = factoryV[0] > fV[0] ||
-		      factoryV[0] == fV[0] && factoryV[1] > fV[1] ||
-		      factoryV[0] == fV[0] && factoryV[1] == fV[1] && factoryV[2] >= fV[2];
-		fprintf(msg,"(factory-%s %s %d.%d.%d) ", PACKAGE_VERSION, fOK ? ">=" : "<", fV[0], fV[1], fV[2]);
-		return fOK ? 0 : 1;
-		}
-	   ]])],
-	[AC_MSG_RESULT(yes)],
-	[AC_MSG_RESULT([no, will build factory ]); BUILD_factory=yes LIBS=$SAVELIBS])
-   break
-done
-test $BUILD_factory = yes && BUILTLIBS="$SINGULARLIBS $BUILTLIBS"
 
 # we need to do the fortran library testing last, in case AC_SEARCH_LIBS adds
 # one of them to $LIBS, making it impossible to check for the presence of C or
